docs/DeveloperGuide.md

```diff
@@ -182,12 +183,57 @@ This section describes some noteworthy details on how certain features are imple
 
 ##### Implementation
 
+#### 2.3 Show event
+
+#### Implementation
+
+The following is a detailed explanation of the operations which take place when the `showevents` command is called
+
+1. After successfully parsing of user input the `ShowEventCommand#exeucte(Model model)` method is called.
+2. Since all events are stored in a `FilteredList`, a `PREDICATE_SHOW_ALL_EVENTS` is passed to the list so that it now contains all the events in the addressbook. The `PREDICATE_SHOW_EVENTS` essentially returns `true` for every entry in the `FilteredList` i.e no item is being filtered out, thus all the events in the addressbook can be accessed through the list.
+3. The GUI contains a pointer to the `FilteredList` present in the `ModelManager`, thus it is able to retrieve all the events present in the addressbook and represent them as a `EventCard` class which is a JavaFX UI feature which visually depicts the event's name, date and time, description and the names of any friends associated with the event.
+4. The `MainWindow` class receives the `CommandResult` after `ShowEventsCommand::execute` is done executing, a boolean `event` within `CommandResult` decides whether `MainWindow::changeInterface` is called which changes the tab from `Friends` tab to `Events` or vice versa.
+
+#### Design Considerations
+
+- Current Implementation
+  - The current implementation fits very well with AB3 and is backward compatible as well. 
+
+- Alternate Implementations
+  - Another possibility is to have a list representing which commands refer to the Friends tab and which commands refer to the Events tab, this list can be checked once a command is entered and the tab can accordingly be switched. However, this implementation involves a lot of maintenance as everytime a new commands is created it will need to be added here, thus we did not choose to proceed with this implementation.
+
+#### 2.4 Edit event
+
+##### Implementation
```

Comment:
I wonder if we can generalize this for all edit commands...cause it's the same for Logs and Persons, no?

Reply:
I agree, but I think for this iteration we allow for the duplication, and once we have everyone's updated DG we can extract out the repeating elements and generalize it.